### DISCUSSION TITLE
Slow startup due to Loading EGL on my laptop with arch linux

### DISCUSSION THREAD

**Discussion by lakinwecker** [wezterm/wezterm#845] [Q&A]
The important part from the logs is this: 
```
2021-06-02T14:08:24.753Z TRACE window::egl                        &gt; load_egl: Library@0x55b49651a030
2021-06-02T14:08:27.792Z TRACE window::egl                        &gt; initialized EGL version 1.5
```
https://gist.github.com/lakinwecker/f97c33f60fb5197ed55801391866ae0a

This is a Dell XPS 15 with nvidia and integrated GPU, but I don't use the nvidia card at all.  Other programs I use don't seem to have this issue, notably alacritty starts up very quickly.

**Comment by wez** [comment]
This sounds like an issue with your EGL installation; the first log line is where wezterm resolves and dlopens the EGL library, the second is printed after it calls the initialize function in that library.  The time elapsed is spent by EGL itself and is outside the control of wezterm.

I'd suggest checking to make sure that you don't have eg: drivers for graphics cards that are not present on your system installed.
You might find that setting some mesa environment variables before starting wezterm will show some additional context:
https://mesa-docs.readthedocs.io/en/latest/envvars.html
https://mesa-docs.readthedocs.io/en/latest/egl.html

I'd suggest trying: `EGL_LOG_LEVEL=debug` in particular!

**Reply by lakinwecker** [reply]
Thanks for the response, I'll dig into it more.  I fully agree that it seems like something is odd on this system.

**Reply by lakinwecker** [reply]
Removing the nvidia driver did not improve the situation, and the `EGL_LOG_LEVEL=debug` didn't elucidate the situation either. Regardless, I appreciate the prompt response.
